pkg/util/topology_test.go

```diff
@@ -0,0 +1,379 @@
+	}
+
+	_, _, _, err = FindPoolAndTopology(&validMultipleTopoPools, &partialLowerSegmentAccReq)
+	if err == nil {
+		t.Errorf("expected failure due to partial segments in accessibility requirements")
+	}
+
+	_, _, _, err = FindPoolAndTopology(&validMultipleTopoPools, &differentSegmentAccReq)
+	if err == nil {
+		t.Errorf("expected failure due to mismatching segments in accessibility requirements")
+	}
+
+	// Test success cases
+	// If a pool is a superset of domains (either empty domain labels or partial), it can be selected
+	poolName, _, topoSegment, err := FindPoolAndTopology(&emptyDomainsInTopoPools, &validAccReq)
+	err = checkOutput(err, poolName, topoSegment)
+	if err != nil {
+		t.Errorf("expected success got: (%v)", err)
+	}
+
+	poolName, _, topoSegment, err = FindPoolAndTopology(&partialDomainsInTopoPools, &validAccReq)
+	err = checkOutput(err, poolName, topoSegment)
+	if err != nil {
+		t.Errorf("expected success got: (%v)", err)
+	}
+
+	// match in a singleton topology pools
+	poolName, _, topoSegment, err = FindPoolAndTopology(&validSingletonTopoPools, &validAccReq)
+	err = checkOutput(err, poolName, topoSegment)
+	if err != nil {
+		t.Errorf("expected success got: (%v)", err)
+	}
+
+	// match first in multiple topology pools
+	poolName, _, topoSegment, err = FindPoolAndTopology(&validMultipleTopoPools, &validAccReq)
+	err = checkOutput(err, poolName, topoSegment)
+	if err != nil {
+		t.Errorf("expected success got: (%v)", err)
+	}
+
+	// match non-first in multiple topology pools
+	switchPoolOrder := []TopologyConstrainedPool{}
+	switchPoolOrder = append(switchPoolOrder, validMultipleTopoPools[1], validMultipleTopoPools[0])
+	poolName, _, topoSegment, err = FindPoolAndTopology(&switchPoolOrder, &validAccReq)
+	err = checkOutput(err, poolName, topoSegment)
+	if err != nil {
+		t.Errorf("expected success got: (%v)", err)
+	}
+
+	// test valid dataPool return
+	for i := range switchPoolOrder {
+		switchPoolOrder[i].DataPoolName = "ec-" + switchPoolOrder[i].PoolName
+	}
+	poolName, dataPoolName, topoSegment, err := FindPoolAndTopology(&switchPoolOrder, &validAccReq)
+	err = checkOutput(err, poolName, topoSegment)
+	if err != nil {
+		t.Errorf("expected success got: (%v)", err)
+	}
+	if dataPoolName != "ec-"+poolName {
+		t.Errorf("expected data pool to be named ec-%s, got %s", poolName, dataPoolName)
+	}
+
+	// TEST: MatchTopologyForPool
+	// check for non-existent pool
+	_, err = MatchTopologyForPool(&validMultipleTopoPools, &validAccReq, pool1+"fuzz")
+	if err == nil {
+		t.Errorf("expected failure due to non-existent pool name (%s) got success", pool1+"fuzz")
+	}
+
+	// check for existing pool
+	topoSegment, err = MatchTopologyForPool(&validMultipleTopoPools, &validAccReq, pool1)
+	err = checkOutput(err, pool1, topoSegment)
+	if err != nil {
+		t.Errorf("expected success got: (%v)", err)
+	}
+}
+
+/*
+// TODO: To test GetTopologyFromDomainLabels we need it to accept a k8s client interface, to mock k8sGetNdeLabels output
```

Comment:
@ShyamsundarR is this TODO planned for follow up PR ?

Reply:
It is left in here if decided to take this up as an InitContainer.

Reply:
IOW, no followup PR planned for this.